scripts/generate_context_bullet.py

```
@@ -18,114 +18,121 @@
                 else:
                     # Fallback: read as text
-                    content = poml_file.read_text(encoding='utf-8')
+                    content = poml_file.read_text(encoding="utf-8")
                     self.templates[poml_file.stem] = content
             except Exception as e:
                 print(f"Warning: Could not load template {poml_file}: {e}")
-    
+
     def generate_context_bullet(
         self,
         data_source: str = None,
```

Ruff (0.13.3)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)